_data-prepper/pipelines/configuration/sinks/s3.md

```diff
@@ -69,121 +71,140 @@ In order to use the `s3` sink, configure AWS Identity and Access Management (IAM
 
+Option | Required                           | Type   | Description
+:--- |:-----------------------------------|:-------| :---
+`flush_capacity_ratio` | No                                 | Float  | The percentage of groups to be force-flushed when `aggregate_threshold maximum_size` is reached. Percentage is expressed from `0.0`--`1.0`. Default is `0.5`.
+`maximum_size` | Yes  | String | The maximum number of bytes to accumulate before force-flushing objects. For example, `128mb`.
 
 ## Buffer type
 
-`buffer_type` is an optional configuration that determines how Data Prepper temporarily stores data before writing an object to S3. The default value is `in_memory`. Use one of the following options:
+`buffer_type` is an optional configuration that determines how Data Prepper temporarily stores data before writing an object to S3. The default value is `in_memory`. 
+
+Use one of the following options:
 
 - `in_memory`: Stores the record in memory.
-- `local_file`: Flushes the record into a file on your local machine. This uses your machine's temporary directory.
+- `local_file`: Flushes the record into a file on your local machine. This option uses your machine's temporary directory.
 - `multipart`: Writes using the [S3 multipart upload](https://docs.aws.amazon.com/AmazonS3/latest/userguide/mpuoverview.html). Every 10 MB is written as a part.
 
 ## Object key configuration
 
+Use the following options to define how object keys are constructed for objects stored in S3.
+
 Option | Required | Type | Description
 :--- | :--- | :--- | :---
-`path_prefix` | No | String | The S3 key prefix path to use for objects written to S3. Accepts date-time formatting. For example, you can use `%{yyyy}/%{MM}/%{dd}/%{HH}/` to create hourly folders in S3. The prefix path should end with `/`. By default, Data Prepper writes objects to the root of the S3 bucket.
-
+`path_prefix` | No | String | The S3 key prefix path to use for objects written to S3. Accepts date-time formatting and dynamic injection of values using [Data Prepper expressions](https://opensearch.org/docs/latest/data-prepper/pipelines/expression-syntax/). For example, you can use `/${/my_partition_key}/%{yyyy}/%{MM}/%{dd}/%{HH}/` to create hourly folders in S3 based on the `my_partition_key` value. The prefix path should end with `/`. By default, Data Prepper writes objects to the S3 bucket root.
 
-## codec
+## `codec`
 
 The `codec` determines how the `s3` source formats data written to each S3 object.
 
-### avro codec
+### `avro` codec
 
-The `avro` codec writes an event as an [Apache Avro](https://avro.apache.org/) document.
+The `avro` codec writes an event as an [Apache Avro](https://avro.apache.org/) document. Because Avro requires a schema, you may either define the schema or have Data Prepper automatically generate it. Defining your own schema is recommended, as this will allow it to be tailored to your particular use case.
```

Comment:
```suggestion
The `avro` codec writes an event as an [Apache Avro](https://avro.apache.org/) document. Because Avro requires a schema, you may either define the schema or have Data Prepper automatically generate it. Defining your own schema is recommended because this will allow it to be tailored to your particular use case.
```